internal/agent/device/spec/manager.go

```diff
@@ -142,6 +163,12 @@ func (s *manager) Ensure() error {
 	// add the desired spec to the queue this ensures that the device will
 	// reconcile the desired spec on startup.
 	s.queue.Add(context.TODO(), desired)
+
+	// Log bootstrap event if this is the first time initializing (all 3 files were just created)
+	if isBootstrap {
+		s.logAudit(context.TODO(), nil, "", "0", audit.AuditResultSuccess, audit.AuditReasonBootstrap, audit.AuditTypeCurrent)
```

Comment:
the auditing should happen at a lower level `s.write` level. the complexity of gathering the information here is not actually needed if all we care about is did the file change. If the file does not exist can we  assume we are bootstrapping it to version 0?

```go
   if err := s.write(specType, newVersionedDevice("0"), reason); err != nil {
```